src/components/input-field/input-field.tsx

```diff
@@ -321,21 +321,26 @@ export class InputField {
         }
 
         return [
-            <label class={this.getContainerClassList()}>
-                <span class="mdc-notched-outline" tabindex="-1">
-                    <span class="mdc-notched-outline__leading"></span>
-                    {this.renderLabel()}
-                    <span class="mdc-notched-outline__trailing"></span>
-                </span>
-                {this.renderLeadingIcon()}
-                {this.renderEmptyValueForReadonly()}
-                {this.renderPrefix()}
-                {this.renderFormattedNumber()}
-                {this.renderInput(properties)}
-                {this.renderSuffix()}
-                {this.renderTextarea(properties)}
-                {this.renderTrailingLinkOrButton()}
-            </label>,
+            <limel-notched-outline
+                labelId={this.labelId}
+                label={this.label}
+                required={this.required}
+                invalid={this.invalid || this.isInvalid()}
+                disabled={this.disabled}
+                readonly={this.readonly}
+                hasValue={!!this.value}
+                hasLeadingIcon={!!this.leadingIcon}
+            >
+                <label slot="content" class={this.getContainerClassList()}>
+                    {this.renderLeadingIcon()}
```

Comment:
Please check this video:

https://github.com/user-attachments/assets/e8a5931d-dd2e-4fdc-9c85-6a7d68c33c2b

When component has a leading icon floating label gets extra space to the left